config/routes.rb

```diff
@@ -6,6 +6,9 @@
 
   resources :quizzes do
     resources :questions
+    get  "leaderboard/create", to: "leaderboards#create", on: :member, as: :leaderboard_create
```

Comment:
this should probably be within a nested resource i.e. so it's inline with the rest of your routes since you're using resources elsewhere

```
resources :leaderboard_entries, only: %i[index create], on: :member

```

Reply:
One issue that this will create is that the create endpoint will be a POST (all create endpoints should be POSTs not GETs)

GETs retrieve something POSTs create something. You might want to disregard this for now since you won't be able to redirect to a POST endpoint from the results controller, but the fact your create is GET is a bit of a code smell and suggests somethings slightly wrong with the flow.

I think your question endpoint should figure out whether it's the final question.
If that is the case then it should post via a form to the leaderboard create endpoint. If not then it can submit to the submit endpoint.

But yea ignore this for now and let's get your PR across the line

Reply:
Cool i think it'd be quite straightforward to do. You could add a method to question like Question#last_question? which uses the same logic you've got atm in the results endpoint, but slightly tweaked 

so instead of 

```
 next_question = @quiz.questions.where("id > ?", current_question.id).first

```

you could have

```
class Question
  def last_question?
    quiz.questions.order(id: :desc).last == self
  end
end
```

or 

```
class Question
  def last_question?
    quiz.questions.where("id > ?", current_question.id).blank?
  end
end
```

there's a bunch of ways you could do it. But yea you can then figure out which endpoint to hit based on that